rkt/api_service.go

```diff
@@ -709,22 +801,25 @@ func (s *v1AlphaAPIServer) ListImages(ctx context.Context, request *v1alpha.List
 }
 
 // getImageInfo for a given image ID, returns the *v1alpha.Image object.
-func getImageInfo(store *imagestore.Store, imageID string) (*v1alpha.Image, error) {
-	key, err := store.ResolveKey(imageID)
-	if err != nil {
-		stderr.PrintE(fmt.Sprintf("failed to resolve the image ID %q", imageID), err)
-		return nil, err
+func getImageInfo(st *imagestore.Store, imageID string) (*v1alpha.Image, error) {
+	var err error
+	if len(imageID) != imagestore.Keylen {
+		imageID, err = st.ResolveKey(imageID)
```

Comment:
I'm not sure I'm understanding this (and the check just above). Are you re-using imageID to implicitly hold different kinds of information? Full-SHA and truncated-SHA? If so, can we make the two codepaths explicit instead of trying to guess input format?